core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
@KostyaSha 
As a **core contributor** with pretty **much contributions** I think your comments are totally invalid:
* the initial comment didn't contain any feedback regarding the code, hence @kzantow 's comment was totally valid
* your offensiveness is not constructive here

I also think the code is good enough